Frontend/ui-driver/src/Flow.purs

```diff
@@ -190,6 +191,13 @@ baseAppFlow baseFlow event = do
       void $ lift $ lift $ setLogField "bundle_version" $ encode (bundle)
       void $ lift $ lift $ setLogField "config_version" $ encode config
       void $ lift $ lift $ setLogField "platform" $ encode (os)
+      when ((getValueToLocalStore T_AND_C_VERSION == "__failed") || (getValueToLocalStore T_AND_C_VERSION == "(null)")) $ do
+        setValueToLocalStore T_AND_C_VERSION "1"
+      let isTermsUpdated = fromMaybe 0 (fromString $ getValueToLocalStore T_AND_C_VERSION) < appConfig.termsVersion
+      if isTermsUpdated then do
```

Comment:
Can you use when

Reply:
in flow file when was not working correctly everytime, so added `if`